### ISSUE TITLE
Results_3d

### ISSUE THREAD

**Issue by ShadiMalekiArasi** [ShadiMalekiArasi/my-master-thesis#13]
Saving logs to output/test_logs/19122020-130254

Configurations:
BATCH_SIZE                     64
CUSTOM_REGRESSOR_IDX           {0: 'regressor_toes_right.npy', 7: 'regressor_toes_left.npy'}
CUSTOM_REGRESSOR_PATH          /content/drive/My Drive/main/models/regressors
DATASETS                       ['mpii_3d']
DATA_DIR                       /content/drive/My Drive/main/output/tfrecords_with_toes
DISCRIMINATOR_LEARNING_RATE    0.0001
DISCRIMINATOR_LOSS_WEIGHT      1
DISCRIMINATOR_WEIGHT_DECAY     0.0001
DS_KP2D                        {'lsp': 14, 'cocoplus': 19, 'custom': 21}
ENCODER_INPUT_SHAPE            (224, 224, 3)
ENCODER_ONLY                   True
EPOCHS                         55
GENERATOR_2D_LOSS_WEIGHT       60.0
GENERATOR_3D_LOSS_WEIGHT       60.0
GENERATOR_LEARNING_RATE        1e-05
GENERATOR_WEIGHT_DECAY         0.0001
ITERATIONS                     3
JOINT_TYPE                     custom
LOG_DIR                        output/test_logs/19122020-130254
NUM_CAMERA_PARAMS              3
NUM_JOINTS                     23
NUM_JOINTS_GLOBAL              24
NUM_KP2D                       21
NUM_KP3D                       16
NUM_PARALLEL                   16
NUM_POSE_PARAMS                72
NUM_SHAPE_PARAMS               10
NUM_SMPL_PARAMS                85
NUM_TEST_SAMPLES               2874
NUM_TRAINING_SAMPLES           166311
NUM_TRAIN_SMPL_SAMPLES         4116233
NUM_VALIDATION_SAMPLES         0
NUM_VERTICES                   6890
RESTORE_PATH                   None
ROOT_DATA_DIR                  output
ROOT_PROJECT_DIR               /content/drive
SCALE_MAX                      1.23
SCALE_MIN                      0.8
SEED                           42
SMPL_DATASETS                  ['cmu', 'joint_lim']
SMPL_DATA_DIR                  /content/drive/My Drive/main/output/tfrecords/smpl
SMPL_MEAN_THETA_PATH           /content/drive/My Drive/main/models/neutral_smpl_mean_params.h5
SMPL_MODEL_PATH                /content/drive/My Drive/main/models/neutral_smpl_coco_regressor.pkl
TRANS_MAX                      20
USE_3D                         True
_Config__initialized           True
_Config__instance              <main.config.Config object at 0x7fcb282c7da0>

building model...

Downloading data from https://storage.googleapis.com/tensorflow/keras-applications/densenet/densenet121_weights_tf_dim_ordering_tf_kernels_notop.h5
29089792/29084464 [==============================] - 0s 0us/step
initialize train dataset...
['/content/drive/My Drive/main/output/tfrecords_with_toes/mpii_3d/*_train.tfrecord']
Done (t=2.085574150085449)

initialize smpl dataset...
['/content/drive/My Drive/main/output/tfrecords/smpl/cmu/*_train.tfrecord', '/content/drive/My Drive/main/output/tfrecords/smpl/joint_lim/*_train.tfrecord']
Done (t=0.6652364730834961)

initialize val dataset...
['/content/drive/My Drive/main/output/tfrecords_with_toes/mpii_3d/*_val.tfrecord']

training:   0%|          | 0/2598 [00:00<?, ?it/s]

Done (t=0.21265769004821777)

Start of Epoch 1
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/backend.py:434: UserWarning: `tf.keras.backend.set_learning_phase` is deprecated and will be removed after 2020-10-11. To update it, simply pass a True/False value to the `training` argument of the `__call__` method of your layer or model.
  warnings.warn('`tf.keras.backend.set_learning_phase` is deprecated and '
training:   2%|▏         | 58/2598 [37:07<26:40:17, 37.80s/it]

Expected begin[0] in [0, 479], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [52:14<26:42:49, 38.22s/it]

Generator Loss: 63.11809158325195, Discriminator Loss: 26.485273361206055

validate: 8it [00:55,  6.91s/it]

MPJPE kp2d: 0.2533341944217682
MPJPE kp3d: 0.893287181854248, MPJPE kp3d aligned: 0.8790944218635559
Time taken for epoch 1 is 3189.802266359329 sec

Start of Epoch 2
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [36:38<26:42:20, 37.85s/it]

Expected begin[0] in [0, 501], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [51:45<26:28:17, 37.88s/it]

Generator Loss: 55.303062438964844, Discriminator Loss: 23.964447021484375

validate: 8it [00:47,  5.97s/it]

MPJPE kp2d: 0.25051531195640564
MPJPE kp3d: 0.8850423097610474, MPJPE kp3d aligned: 0.8690481781959534
Time taken for epoch 2 is 3153.888743877411 sec

Start of Epoch 3
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [36:51<26:44:15, 37.90s/it]

Expected begin[0] in [0, 481], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [51:44<26:27:48, 37.87s/it]

Generator Loss: 52.95711898803711, Discriminator Loss: 21.77948760986328

validate: 8it [00:47,  5.90s/it]

MPJPE kp2d: 0.2546297311782837
MPJPE kp3d: 0.8800233006477356, MPJPE kp3d aligned: 0.8691135048866272
Time taken for epoch 3 is 3152.847270011902 sec

Start of Epoch 4
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [36:00<26:09:46, 37.08s/it]

Expected begin[0] in [0, 466], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [50:44<25:56:57, 37.13s/it]

Generator Loss: 51.69675827026367, Discriminator Loss: 19.72880744934082

validate: 8it [00:46,  5.87s/it]

MPJPE kp2d: 0.257983922958374
MPJPE kp3d: 0.8815485835075378, MPJPE kp3d aligned: 0.8719890117645264
Time taken for epoch 4 is 3093.093432188034 sec

Start of Epoch 5
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [36:02<26:18:48, 37.29s/it]

Expected begin[0] in [0, 474], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [51:00<26:04:53, 37.32s/it]

Generator Loss: 50.72148132324219, Discriminator Loss: 17.774913787841797

validate: 8it [00:47,  5.88s/it]

MPJPE kp2d: 0.25430938601493835
MPJPE kp3d: 0.8831079006195068, MPJPE kp3d aligned: 0.875835657119751
Time taken for epoch 5 is 3108.346257209778 sec

saving checkpoint

Start of Epoch 6
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [36:07<26:19:16, 37.31s/it]

Expected begin[0] in [0, 496], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [51:01<26:05:45, 37.34s/it]

Generator Loss: 49.738399505615234, Discriminator Loss: 16.05328369140625

validate: 8it [00:47,  5.88s/it]

MPJPE kp2d: 0.2673419713973999
MPJPE kp3d: 0.8874866366386414, MPJPE kp3d aligned: 0.8831024169921875
Time taken for epoch 6 is 3110.224895954132 sec

Start of Epoch 7
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [35:59<26:05:00, 36.97s/it]

Expected begin[0] in [0, 494], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [50:50<26:00:11, 37.21s/it]

Generator Loss: 48.45083236694336, Discriminator Loss: 14.676104545593262

validate: 8it [00:47,  5.93s/it]

MPJPE kp2d: 0.2580413222312927
MPJPE kp3d: 0.8948310613632202, MPJPE kp3d aligned: 0.8928284645080566
Time taken for epoch 7 is 3098.82789683342 sec

Start of Epoch 8
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [35:53<26:05:25, 36.98s/it]

Expected begin[0] in [0, 493], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [50:39<25:54:25, 37.07s/it]

Generator Loss: 47.08543014526367, Discriminator Loss: 13.671699523925781

validate: 8it [00:46,  5.86s/it]

MPJPE kp2d: 0.2561356723308563
MPJPE kp3d: 0.9060412645339966, MPJPE kp3d aligned: 0.9054877161979675
Time taken for epoch 8 is 3086.8694920539856 sec

Start of Epoch 9
<PrefetchDataset shapes: ((64, 224, 224, 3), (64, 21, 3), (64, 16, 3), (64,)), types: (tf.float32, tf.float32, tf.float32, tf.int64)>

training:   2%|▏         | 58/2598 [35:53<26:05:31, 36.98s/it]

Expected begin[0] in [0, 503], but got -2147483578
	 [[{{node Slice}}]]

training:   3%|▎         | 82/2598 [50:42<25:56:04, 37.11s/it]

Generator Loss: 45.78561782836914, Discriminator Loss: 13.08407211303711

validate: 8it [00:47,  5.88s/it]

MPJPE kp2d: 0.2521296441555023
MPJPE kp3d: 0.9184232354164124, MPJPE kp3d aligned: 0.9160627126693726
Time taken for epoch 9 is 3090.229208469391 sec
